lib/dns/dns.go

```diff
@@ -320,6 +320,48 @@ Redo:
 	return true, nil
 }
 
+// lookupByAlias looks up a container by alias, given the requesting container
+// and the network of the incoming request. It first does a container-scoped
+// alias search, followed by a network-scoped aliased search for alias. The
+// result is a collection of endpoints for the matching container that are
+// only on the same network as the requesting container.
+func lookupByAlias(ctx *network.Context, scope *network.Scope, reqc *network.Container, alias string) []*network.Endpoint {
+	// container specific alias search
+	cons := ctx.ContainersByAlias(fmt.Sprintf("%s:%s:%s", scope.Name(), reqc.Name(), alias))
```

Comment:
This string looks to follow a well defined format (`<scopeName>:<containerName>:<string alias>`).  Can this format be hidden behind a helper that returns the result you want but doesnt require `Sprintf`?